File: src/SIStatisticsService.Contract/Models/PackageInfo.cs
namespace SIStatisticsService.Contract.Models;

/// <summary>
/// Represents a game package info.
/// </summary>
/// <param name="Name">Package name.</param>
/// <param name="Hash">Package hash.</param>
/// <param name="Authors">Package authors.</param>
/// <param name="AuthorsContacts">Package author contacts.</param>
public sealed record PackageInfo(string? Name, string? Hash, string[] Authors, string? AuthorsContacts = null);

File: web/src/models/PackageInfo.ts
/** Represents a game package info. */
export default interface PackageInfo {
	/** Game package name. */
	name: string;

	/** Game package hash. */
	hash: string;

	/** Game package authors. */
	authors: string[];

	/** Package author contacts. */
	authorsContacts: string;

	/** Package source URI. */
	source?: string;
}